misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -79,4 +87,75 @@ static void sortImportDeclarations(List<ImportDeclarationNode> importDeclaration
+            // remove comments from the previous first import
+            ImportDeclarationNode prevFirstImportNode = importNodes.get(prevFirstIndex);
+            MinutiaeList prevLeadingMinutiae = prevFirstImportNode.leadingMinutiae();
+            List<Minutiae> leadingMinutiae = new ArrayList<>();
+            Minutiae prevFirstMinutiae = prevLeadingMinutiae.get(0);
+            if (prevFirstMinutiae.kind() != SyntaxKind.COMMENT_MINUTIAE) {
+                leadingMinutiae.add(prevFirstMinutiae);
+            }
+
+            Token prevFirstImportToken = prevFirstImportNode.importKeyword();
+            Token modifiedPrevFirstImportToken =
+                    prevFirstImportToken.modify(NodeFactory.createMinutiaeList(leadingMinutiae),
+                            prevFirstImportToken.trailingMinutiae());
+            importNodes.set(prevFirstIndex,
+                    prevFirstImportNode.modify().withImportKeyword(modifiedPrevFirstImportToken).apply());
+
+            // add leading comments from the previous first import
+            prevLeadingMinutiae = firstImportNode.leadingMinutiae();
+            prevFirstMinutiae = prevLeadingMinutiae.get(0);
+            if (prevFirstMinutiae.kind() == SyntaxKind.END_OF_LINE_MINUTIAE) {
+                prevLeadingMinutiae = prevLeadingMinutiae.remove(0);
+            }
+            if (!hasEmptyLineAtEnd(prevLeadingMinutiae)) {
+                prevLeadingMinutiae =
+                        prevLeadingMinutiae.add(NodeFactory.createEndOfLineMinutiae(System.lineSeparator()));
+            }
+            ImportDeclarationNode sortedFirstImportNode = importNodes.get(0);
+            Token sortedFirstImportToken = sortedFirstImportNode.importKeyword();
+            MinutiaeList sortedLeadingMinutiae = sortedFirstImportToken.leadingMinutiae();
+            for (Minutiae minutiae : sortedLeadingMinutiae) {
+                prevLeadingMinutiae = prevLeadingMinutiae.add(minutiae);
+            }
+
+            Token modifiedSortedFirstImportToken =
+                    sortedFirstImportToken.modify(prevLeadingMinutiae, sortedFirstImportToken.trailingMinutiae());
+            importNodes.set(0,
+                    sortedFirstImportNode.modify().withImportKeyword(modifiedSortedFirstImportToken).apply());
+        }
+    }
+
+    private static String getImportString(ImportDeclarationNode importDeclarationNode) {
+        Optional<ImportOrgNameNode> orgNameNode = importDeclarationNode.orgName();
+        String orgName = orgNameNode.isPresent() ? orgNameNode.get().toSourceCode() : "";
+        String moduleName = importDeclarationNode.moduleName().stream()
+                .map(Node::toSourceCode)
+                .collect(Collectors.joining("."));
+        return orgName + moduleName;
+    }
+
+    private static boolean hasEmptyLineAtEnd(MinutiaeList minutiaeList) {
+        int size = minutiaeList.size();
+        return minutiaeList.get(size - 1).kind() == SyntaxKind.END_OF_LINE_MINUTIAE &&
```

Comment:
Is the size always guaranteed to be > 1?

Reply:
We go into this check when the there is a comment minutiae attached with the import node. When a comment minutiae is attached and a newline is attached to the node. Thus it is guaranteed to be > 1.